src/main/java/core/basesyntax/db/ShopStorage.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.db;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.OperationType;
+import java.util.Set;
+
+public interface ShopStorage<T> {
+    int getAmount(T t);
+
+    void save(Fruit fruit, int value, OperationType operationType);
```

Comment:
The `save` method should use the generic type `T` instead of `Fruit` to maintain consistency with the rest of the interface. Change the method signature to `void save(T t, int value, OperationType operationType);` to ensure the interface can handle any type `T`.